client/src/app/tabs/form/linting/__tests__/FormLinterSpec.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.
+ *
+ * Camunda licenses this file to you under the MIT; you may not use this file
+ * except in compliance with the MIT License.
+ */
+
+import FormLinter from '../FormLinter';
+
+import camundaCloud10 from './camunda-cloud-1-0.json';
+import camundaCloud10Errors from './camunda-cloud-1-0-errors.json';
+import camundaCloud11 from './camunda-cloud-1-1.json';
+import camundaPlatform715 from './camunda-platform-7-15.json';
+import camundaPlatform715Errors from './camunda-platform-7-15-errors.json';
+import noEngineProfile from './no-engine-profile.json';
+
+
+describe('FormLinter', function() {
+
+  it('should lint (JSON string)', function() {
+
+    // when
+    const results = FormLinter.lint(JSON.stringify(camundaPlatform715));
+
+    // then
+    expect(results).to.exist;
+    expect(results).to.be.empty;
+  });
+
+
+  it('should lint (JSON object)', function() {
+
+    // when
+    const results = FormLinter.lint(camundaPlatform715);
+
+    // then
+    expect(results).to.exist;
+    expect(results).to.be.empty;
+  });
+
+
+  it('should not lint if no engine profile', function() {
+
+    // when
+    const results = FormLinter.lint(noEngineProfile);
+
+    // then
+    expect(results).to.exist;
+    expect(results).to.be.empty;
+  });
+
+
+  describe('Camunda Platform 7.15', function() {
+
+    it('should lint without errors', function() {
+
+      // when
+      const results = FormLinter.lint(camundaPlatform715);
+
+      // then
+      expect(results).to.exist;
+      expect(results).to.be.empty;
+    });
+
+
+    it('should lint with errors', function() {
+
+      // when
+      const results = FormLinter.lint(camundaPlatform715Errors);
+
+      // then
+      expect(results).to.exist;
+      expect(results).to.eql([
+        {
```

Comment:
Is this the same representation that we use in bpmnlint (cf. https://github.com/camunda/camunda-modeler/issues/2325)?

Reply:
@philippfromme

Reply:
In bpmnlint the results are grouped by rule name e.g.

```javascript
{
  'foo-rule': [
    ...
  ],
  'bar-rule': [
    ...
  ]
}
```

As we don't have multiple rules for forms at the moment I kept it simple. Happy to change it, if you think it makes sense.

Reply:
Fine with me. We'd need to account for that as a next step, i.e. immediately when tackling BPMN linting.